forgerock-authenticator/src/main/java/org/forgerock/android/authenticator/Notification.java

```diff
@@ -143,6 +146,68 @@ public final boolean isPending() {
         return pending;
     }
 
+    @Override
+    public String toJson() {
+        JSONObject jsonObject = new JSONObject();
+        try {
+            jsonObject.put("id", getId());
+            jsonObject.put("mechanismUID", getMechanismUID());
+            jsonObject.put("messageId", getMessageId());
+            jsonObject.put("challenge", getChallenge());
+            jsonObject.put("amlbCookie", getAmlbCookie());
+            jsonObject.put("timeAdded", getTimeAdded() != null ? getTimeAdded().getTimeInMillis() : null);
+            jsonObject.put("timeExpired", getTimeExpired() != null ? getTimeExpired().getTimeInMillis() : null);
+            jsonObject.put("ttl", getTtl());
+            jsonObject.put("approved", isApproved());
+            jsonObject.put("pending", isPending());
+        } catch (JSONException e) {
+            throw new RuntimeException("Error parsing Notification object to JSON string representation.", e);
+        }
+        return jsonObject.toString();
+    }
+
+    /**
+     * Deserializes the specified Json into an object of the {@link Notification} object.
+     * @param jsonString the json string representing the object to be deserialized
+     * @return a {@link Notification} object from the string. Returns {@code null} if {@code jsonString} is {@code null}
+     * or if {@code jsonString} is empty.
+     */
+    public static Notification fromJson(String jsonString) {
+        if (jsonString == null || jsonString.length() == 0) {
+            return null;
+        }
+
+        try {
+            JSONObject jsonObject = new JSONObject(jsonString);
+            return Notification.builder()
+                    .setMechanismUID(jsonObject.getString("mechanismUID"))
+                    .setMessageId(jsonObject.getString("messageId"))
+                    .setChallenge(jsonObject.getString("challenge"))
+                    .setAmlbCookie(jsonObject.has("amlbCookie") ? jsonObject.getString("amlbCookie"): null)
+                    .setTimeAdded(jsonObject.has("timeAdded") ? getDate(jsonObject.optLong("timeAdded")) : null)
+                    .setTimeExpired(jsonObject.has("timeExpired") ? getDate(jsonObject.optLong("timeExpired")) : null)
+                    .setTtl(jsonObject.optLong("ttl", -1))
+                    .setApproved(jsonObject.getBoolean("approved"))
+                    .setPending(jsonObject.getBoolean("pending"))
+                    .build();
+        } catch (JSONException e) {
+            return null;
```

Comment:
Intent to return null if not able to parse it?

Reply:
Yes, it is expected to be null if not able to parse. Updated the method documentation to make it clear.